Parsely/wp-parsely
# Fix cookie parsing -- values can contain equal signs

## Description


Pull in a library to parse out the value for the `_parsely_visitor` cookie instead of rolling our own.

Additionally, pass the `uuid` through `encodeURIComponet` for good measure.

## Motivation and Context



Unfortunately, when the recommended content widget was refactored in #279, the lib that we wrote to parse out cookies neglected to take into account cookie values that contain an equal sign (`=`). I noticed on a site that has personalization enabled, the cookie containing the uuid was not getting parsed. This means that personalization of results in the widget has been broken since version 2.5.

The [js-cookie library](https://www.npmjs.com/package/js-cookie) is active, mature, well-tested, and relatively small. Leveraging it allows us to rely on its battery of testing. Even so, I've adapted the testing for our previous lib to cover the previously covered use cases as well as the issue this PR intends to fix all the same.

## How Has This Been Tested?




### Automated testing

`npm run test`

### Manual testing

* Run this branch on a site with API access
* Configure the widget to display (in a sidebar, footer, etc.)
* Browse to a page with the widget on it
    * Confirm that personalized results are requested / received
    * Namely, you should see a populated `uuid` query argument in the call to `https://api.parsely.com/v2/related` as per [this line](https://github.com/Parsely/wp-parsely/blob/f465d3db4536bddb22b5ffb9388e405a87e240be/src/js/widgets/recommended/index.js#L31).

## Screenshots (if appropriate):

## Types of changes

Bug fix (non-breaking change which fixes an issue)
